itests/snapshot_internal_test.go

```diff
@@ -58,25 +61,46 @@ func (s *SimpleSnapshotSuite) TestSimpleSnapshot() {
 		time.Sleep(delay)
 	}
 
+	err = s.Client.Connection.SubscribeForMessages(
+		reflect.TypeOf(&proto.GetBlockIdsMessage{}),
+		reflect.TypeOf(&proto.GetBlockMessage{}),
+		reflect.TypeOf(&proto.ScoreMessage{}),
+		reflect.TypeOf(&proto.MicroBlockRequestMessage{}),
+	)
+	require.NoError(s.T(), err, "failed to subscribe for messages")
+
 	// Calculate new score and send score to the node.
 	genesisScore := calculateScore(s.Cfg.BlockchainSettings.Genesis.BaseTarget)
 	blockScore := calculateCumulativeScore(genesisScore, bl.BaseTarget)
 	scoreMsg := &proto.ScoreMessage{Score: blockScore.Bytes()}
 	s.Client.Connection.SendMessage(scoreMsg)
-	time.Sleep(100 * time.Millisecond)
+
+	// Wait for the node to request block IDs.
+	_, err = s.Client.Connection.AwaitMessage(reflect.TypeOf(&proto.GetBlockIdsMessage{}), messageTimeout)
+	require.NoError(s.T(), err, "failed to wait for block IDs request")
 
 	// Send block IDs to the node.
 	blocksMsg := &proto.BlockIdsMessage{Blocks: []proto.BlockID{bl.BlockID()}}
 	s.Client.Connection.SendMessage(blocksMsg)
 	time.Sleep(100 * time.Millisecond)
 
+	// Wait for the node to request the block.
+	blockID, err := s.Client.Connection.AwaitGetBlockMessage(messageTimeout)
+	require.NoError(s.T(), err, "failed to wait for block request")
+	assert.Equal(s.T(), bl.BlockID(), blockID)
+
 	// Marshal the block and send it to the node.
 	bb, err := bl.MarshalToProtobuf(s.Cfg.BlockchainSettings.AddressSchemeCharacter)
 	require.NoError(s.T(), err, "failed to marshal block")
 	blMsg := &proto.PBBlockMessage{PBBlockBytes: bb}
 	s.Client.Connection.SendMessage(blMsg)
 
-	// Wait for 2.5 seconds and send mb-block.
+	// Wait for updated score message.
+	score, err := s.Client.Connection.AwaitScoreMessage(messageTimeout)
+	require.NoError(s.T(), err, "failed to wait for score")
+	assert.Equal(s.T(), blockScore, score)
+
+	// Wait for 2.5 seconds and send micro-block.
 	time.Sleep(2500 * time.Millisecond)
```

Comment:
And this one?

Reply:
Yes, this one is to imitate real life. It can be made less to speed up the test.

Reply:
But I think it's not an issue for integration testing.